### PR TITLE
[BottomNavigation] Add MDCElevation support.

### LINKED ISSUE #8026: [BottomNavigation] Internal issue: b/138202857
This was filed as an internal issue. If you are a Googler, please visit [b/138202857](http://b/138202857) for more details.

---
#### Internal data
- Associated internal bug: [b/138202857](http://b/138202857)

### PR DESCRIPTION
**Pull request by romoore**
Adds support for the `MDCElevatable` and `MDCElevationOverriding` protocols.

Closes #8026

### CODE REVIEW

**Comment by yarneo** [comment]
I think this may be blocked on: https://github.com/material-components/material-components-ios/pull/8034

Because we are separating the override property to its own protocol

File: components/BottomNavigation/src/MDCBottomNavigationBar.m [hunk 1]
```diff
@@ -76,6 +76,9 @@ @interface MDCBottomNavigationBar () <MDCInkTouchControllerDelegate,
 
 @implementation MDCBottomNavigationBar
 
+@synthesize mdc_overrideBaseElevation = _mdc_overrideBaseElevation;
```
Comment (yarneo) [inline components/BottomNavigation/src/MDCBottomNavigationBar.m:79]: Isn't it defaulting to the underscore, can't we omit the `= _mdc...` part?

File: components/BottomNavigation/tests/unit/MDCBottomNavigationBarTests.m [hunk 2]
```diff
@@ -560,4 +566,27 @@ - (void)testTraitCollectionDidChangeBlockCalledWhenTraitCollectionChanges {
...
+  CGFloat expectedBaseElevation = 99;
+
+  // When
+  self.bottomNavBar.mdc_overrideBaseElevation = expectedBaseElevation;
+
+  // Then
+  XCTAssertEqualWithAccuracy(self.bottomNavBar.mdc_overrideBaseElevation, expectedBaseElevation,
+                             0.001);
+}
+
```
Comment (yarneo) [inline components/BottomNavigation/tests/unit/MDCBottomNavigationBarTests.m:591]: please add a test for the elevationDidChangeBlock

**Review by yarneo — CHANGES_REQUESTED** [review]

File: components/BottomNavigation/tests/unit/MDCBottomNavigationBarTests.m (hunk 2 above)
Comment (romoore) [reply components/BottomNavigation/tests/unit/MDCBottomNavigationBarTests.m:591]: It's not called by anything - what would you like to test?

File: components/BottomNavigation/src/MDCBottomNavigationBar.m (hunk 1 above)
Comment (romoore) [reply components/BottomNavigation/src/MDCBottomNavigationBar.m:79]: If I don't specify the ivar name, it seemed to synthesize it to `mdc_overrideBaseElevation`. 

Screen Shot 2019-07-23 at 11 24 36 PM

Comment (yarneo) [reply components/BottomNavigation/src/MDCBottomNavigationBar.m:79]: sweet, thank you

File: components/BottomNavigation/tests/unit/MDCBottomNavigationBarTests.m (hunk 2 above)
Comment (yarneo) [reply components/BottomNavigation/tests/unit/MDCBottomNavigationBarTests.m:591]: you can verify that it is correctly stored in the component:
```
- (void)testElevationDidChangeBlockIsCorrectlySetUp {
  // Given
  __block NSNumber *weakElevationDidChangeBlockWasCalled = @NO;
  self.card.mdc_elevationDidChangeBlock = ^(CGFloat elevation) {
    weakElevationDidChangeBlockWasCalled = @YES;
  };

  // When
  self.card.mdc_elevationDidChangeBlock(0);

  // Then
  XCTAssertTrue(weakElevationDidChangeBlockWasCalled.boolValue);
}
```

Comment (romoore) [reply components/BottomNavigation/tests/unit/MDCBottomNavigationBarTests.m:591]: Added functional tests now that the category has been written.

**Review by yarneo — APPROVED** [review]

**Pull request merged**